Microsoft.NET.Build.Containers/CreateNewImage.cs

```diff
@@ -127,15 +149,20 @@ public override bool Execute()
         GeneratedContainerManifest =  JsonSerializer.Serialize(image.manifest);
         GeneratedContainerConfiguration = image.config.ToJsonString();
 
-        Registry? outputReg = IsDockerPush ? null : new Registry(ContainerHelpers.TryExpandRegistryToUri(OutputRegistry));
-        foreach (var tag in ImageTags)
+        
+        foreach (var destinationImageReference in destinationImageReferences)
         {
             if (IsDockerPush)
             {
+                var localDaemon = new LocalDocker(msg => Log.LogMessage(msg));
```

Comment:
I'm looking forward to issues like https://github.com/dotnet/sdk-container-builds/issues/271, where we will want some ability to pivot based on the container runtime the user has chosen (likely one of docker, podman, or nerdctl). Each of these will need these same logical operations, so I figured I would introduce them now. If we have the interface in palce, we can tell an interested contributor:

a) implement this interface for your chosen runtime
b) test it
c) define a 'switching' mechanism for container runtimes and plug yours in

which seems like a much less amount of work than also refactoring away to an interface as well.

In fact, having written that down, I think I might go ahead and do `c)` as part of this PR as well, so that users already have the model to work against and just need to implement their interface.

Reply:
Ok, I just pushed a commit that does `c)`, so now any adding support for a new runtime would mean:
* adding the name of the flag
* implementing the interface
* adding to the switch statement

That is easy enough for contributors, I think.